repos/fdbt-site/src/pages/api/selectSalesOfferPackage.ts

```diff
@@ -22,33 +22,46 @@ const pricePrefix = 'price-';
 
 export const sanitiseReqBody = (req: NextApiRequestWithSession): SanitisedBodyAndErrors => {
     const sanitisedBody: { [key: string]: SalesOfferPackage[] } = {};
+    const keys = Object.keys(req.body);
 
     const errors: ErrorInfo[] = [];
     Object.entries(req.body).forEach((item) => {
         const [key, value]: [string, unknown] = item;
 
-        if (key.startsWith(productPrefix)) {
-            if (value && Array.isArray(value)) {
-                sanitisedBody[key.substring(productPrefix.length)] = value
-                    .filter((a) => a)
-                    .map((it) => JSON.parse(it) as SalesOfferPackage);
+        if (!key.startsWith('0')) {
+            return;
+        }
+
+        const id = key.split('-')[0];
+
+        if (key.startsWith(`${id}-${productPrefix}`)) {
+            if (value) {
+                if (Array.isArray(value)) {
+                    sanitisedBody[key.substring(productPrefix.length + id.length + 1)] = value
+                        .filter((a) => a)
+                        .map((it) => JSON.parse(it) as SalesOfferPackage);
+                } else if (!Array.isArray(value)) {
+                    sanitisedBody[key.substring(productPrefix.length + id.length + 1)] = [
+                        JSON.parse(value as string) as SalesOfferPackage,
+                    ];
+                }
             } else {
+                const keyWithIdRemoved = key.split(`${id}-`)[1];
```

Comment:
'key.replace' would be more correct here I think

Reply:
addressed